platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesWriter.java

```diff
@@ -221,56 +265,46 @@ public Long registerDiscontinuity(final long newOriginRound) {
             logger.error(EXCEPTION.getMarker(), "registerDiscontinuity() called while replaying events");
         }
 
-        final Long latestDurableSequenceNumberUpdate;
-        if (currentMutableFile != null) {
-            closeFile();
-            latestDurableSequenceNumberUpdate = lastWrittenEvent;
-        } else {
-            latestDurableSequenceNumberUpdate = null;
+        try {
+            if (currentMutableFile != null) {
+                closeFile();
+                return lastFlushedEvent;
+            } else {
+                return null;
+            }
+        } finally {
+            fileManager.registerDiscontinuity(newOriginRound);
         }
-
-        fileManager.registerDiscontinuity(newOriginRound);
-
-        return latestDurableSequenceNumberUpdate;
     }
 
     /**
-     * Make sure that the event has a valid stream sequence number.
+     * Submit a request to flush a given sequence number to disk as soon as possible. This flush request will be
+     * honored as soon as the event with the given sequence number is received.
+     *
+     * @param sequenceNumber the sequence number to flush
+     * @return the sequence number of the last event durably written to the stream, or null if this method call didn't
+     * result in any additional events being durably written to the stream
      */
-    private static void validateSequenceNumber(@NonNull final GossipEvent event) {
-        if (event.getStreamSequenceNumber() == GossipEvent.NO_STREAM_SEQUENCE_NUMBER
-                || event.getStreamSequenceNumber() == GossipEvent.STALE_EVENT_STREAM_SEQUENCE_NUMBER) {
-            throw new IllegalStateException("Event must have a valid stream sequence number");
-        }
+    @Nullable
+    public Long submitFlushRequest(final long sequenceNumber) {
```

Comment:
What is this return value used for?

Reply:
If submission of the flush request immediately results in an event becoming durable, this return value will pass that along to the durability nexus